c_glib/arrow-glib/decimal.cpp

```diff
@@ -217,6 +218,125 @@ garrow_decimal128_to_integer(GArrowDecimal128 *decimal)
   return static_cast<int64_t>(*arrow_decimal);
 }
 
+/**
+ * garrow_decimal128_plus:
+ * @left: A #GArrowDecimal128.
+ * @right: A #GArrowDecimal128.
+ *
+ * Returns: (transfer full): The added value of these decimals.
+ *
+ * Since: 0.11.0
+ */
+GArrowDecimal128 *
+garrow_decimal128_plus(GArrowDecimal128 *left,
+                       GArrowDecimal128 *right)
+{
+  auto arrow_decimal_left = garrow_decimal128_get_raw(left);
+  auto arrow_decimal_right = garrow_decimal128_get_raw(right);
+  auto arrow_decimal =
+    std::make_shared<arrow::Decimal128>(*arrow_decimal_left + *arrow_decimal_right);
+  return garrow_decimal128_new_raw(&arrow_decimal);
+}
+
+/**
+ * garrow_decimal128_minus:
+ * @left: A #GArrowDecimal128.
+ * @right: A #GArrowDecimal128.
+ *
+ * Returns: (transfer full): The subtracted value of these decimals.
+ *
+ * Since: 0.11.0
+ */
+GArrowDecimal128 *
+garrow_decimal128_minus(GArrowDecimal128 *left,
+                        GArrowDecimal128 *right)
+{
+  auto arrow_decimal_left = garrow_decimal128_get_raw(left);
+  auto arrow_decimal_right = garrow_decimal128_get_raw(right);
+  auto arrow_decimal =
+    std::make_shared<arrow::Decimal128>(*arrow_decimal_left - *arrow_decimal_right);
+  return garrow_decimal128_new_raw(&arrow_decimal);
+}
+
+/**
+ * garrow_decimal128_multiply:
+ * @left: A #GArrowDecimal128.
+ * @right: A #GArrowDecimal128.
+ *
+ * Returns: (transfer full): The multiplied value of these decimals.
+ *
+ * Since: 0.11.0
+ */
+GArrowDecimal128 *
+garrow_decimal128_multiply(GArrowDecimal128 *left,
+                           GArrowDecimal128 *right)
+{
+  auto arrow_decimal_left = garrow_decimal128_get_raw(left);
+  auto arrow_decimal_right = garrow_decimal128_get_raw(right);
+  auto arrow_decimal =
+    std::make_shared<arrow::Decimal128>(*arrow_decimal_left * *arrow_decimal_right);
+  return garrow_decimal128_new_raw(&arrow_decimal);
+}
+
+/**
+ * garrow_decimal128_divide:
+ * @left: A #GArrowDecimal128.
+ * @right: A #GArrowDecimal128.
+ * @error: (nullable): Return location for a #GError or %NULL.
+ *
+ * Returns: (nullable) (transfer full): The divided value of
+ *   these decimals or %NULL on error.
+ *
+ * Since: 0.11.0
+ */
+GArrowDecimal128 *
+garrow_decimal128_divide(GArrowDecimal128 *left,
+                         GArrowDecimal128 *right,
+                         GError **error)
+{
+  auto arrow_decimal_left = garrow_decimal128_get_raw(left);
+  auto arrow_decimal_right = garrow_decimal128_get_raw(right);
+  arrow::Decimal128 result;
+  arrow::Decimal128 remainder;
+  auto status =
+    arrow_decimal_left->Divide(*arrow_decimal_right, &result, &remainder);
+  if (garrow_error_check(error, status, "[decimal][divide]")) {
+    auto arrow_result = std::make_shared<arrow::Decimal128>(result);
+    return garrow_decimal128_new_raw(&arrow_result);
+  } else {
+    return NULL;
+  }
+}
+
+/**
+ * garrow_decimal128_remainder:
+ * @left: A #GArrowDecimal128.
+ * @right: A #GArrowDecimal128.
+ *
```

Comment:
Can you add an entry for `@error`?

Reply:
Thanks! I'll remove `garrow_decimal128_remainder()`.